### PR TITLE
Implement system contract for `TOKEN_UPDATE_KEYS`

### LINKED ISSUE #8594: UPDATE_TOKEN_KEYS (TokenUpdateKeysPrecompile)

### PR DESCRIPTION
**Pull request by petreze**
**Description**:
Implementation for HTS `updateTokenKeys` function.

Created [UpdateKeysTranslator](https://github.com/hashgraph/hedera-services/blob/08594-hts-update-token-keys/hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/hts/update/UpdateKeysTranslator.java) and its corresponding unit test [UpdateKeysTranslatorTest](https://github.com/hashgraph/hedera-services/blob/08594-hts-update-token-keys/hedera-node/hedera-smart-contract-service-impl/src/test/java/com/hedera/node/app/service/contract/impl/test/exec/systemcontracts/hts/update/UpdateKeysTranslatorTest.java)

Modified [UpdateDecoder](https://github.com/hashgraph/hedera-services/blob/08594-hts-update-token-keys/hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/hts/update/UpdateDecoder.java) to handle `updateTokenKeys` decoding

Added some XTest cases in [UpdateKeysXTest](https://github.com/hashgraph/hedera-services/blob/08594-hts-update-token-keys/hedera-node/hedera-app/src/xtest/java/contract/UpdateKeysXTest.java)

- should successfully update token keys
- should throw `INVALID_ADMIN_KEY` as we are passing an invalid key with key type admin key

**Related issue(s)**:

Fixes #8594 

**Notes for reviewer**:

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

**Review by agadzhalov — DISMISSED** [review]
LGTM!

File: hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/hts/update/UpdateKeysTranslator.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+    @Override
+    public boolean matches(@NonNull HtsCallAttempt attempt) {
+        return Arrays.equals(attempt.selector(), TOKEN_UPDATE_KEYS_FUNCTION.selector());
+    }
+
+    @Override
+    public HtsCall callFrom(@NonNull HtsCallAttempt attempt) {
+        return new DispatchForResponseCodeHtsCall<>(
+                attempt, nominalBodyFor(attempt), SingleTransactionRecordBuilder.class);
```
[the remark this answers is not in the document]
Comment (petreze) [reply hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/hts/update/UpdateKeysTranslator.java]: Agreed, could be replaced

**Review by tinker-michaelj — APPROVED** [review]
LGTM!

**Review by tinker-michaelj — APPROVED** [review]
LGTM!

**Pull request merged**